tests/rancher-istio/rancher_istio.go

```diff
@@ -0,0 +1,947 @@
+				SetValue(
+					"egressGateways.enabled", "false",
+				),
+		},
+
+		{
+			Name: "Set .Values.ingressGateways.type",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				SetValue(
+					"egressGateways.type", "testType",
+				),
+		},
+
+		{
+			Name: "Set .Values.egressGateways.hpaspec",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				Set(
+					"egressGateways.hpaspec", testHPASpec,
+				),
+		},
+
+		{
+			Name: "Set .Values.egressGateways.podDisruptionBudget",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				Set(
+					"egressGateways.podDisruptionBudget", testPodDisruptionBudget,
+				),
+		},
+
+		{
+			Name: "Set .Values.ingressGateways",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				SetValue(
+					"ingressGateways.enabled", "true",
+				).
+				SetValue(
+					"ingressGateways.type", "testType",
+				).
+				Set(
+					"ingressGateways.hpaspec", testHPASpec,
+				).
+				Set(
+					"ingressGateways.podDisruptionBudget", testPodDisruptionBudget,
+				),
+		},
+	},
+
+	NamedChecks: []test.NamedCheck{
+		{
+			Name:   "All Deployments Have ServiceAccount",
+			Checks: common.AllWorkloadsHaveServiceAccount,
+		},
+
+		{
+			Name:   "All Workloads Have Node Selectors and Tolerations",
+			Checks: common.AllWorkloadsHaveNodeSelectorsAndTolerationsForOS,
+		},
+
+		{
+			Name:   "All Workload Container Should Have SystemDefaultRegistry",
+			Checks: common.AllContainerImagesShouldHaveSystemDefaultRegistryPrefix,
+			Covers: []string{
+				".Values.global.cattle.systemDefaultRegistry",
+			},
+		},
+
+		{
+			Name: "Check All Workload Have NodeSelector As Per Given Values",
+			Covers: []string{
+				".Values.nodeSelector",
+			},
+
+			Checks: test.Checks{
+				checker.PerWorkload(func(tc *checker.TestContext, obj metav1.Object, podTemplateSpec corev1.PodTemplateSpec) {
+					nodeselectorAddedByValues, _ := checker.RenderValue[map[string]string](tc, ".Values.nodeSelector")
+					expectedNodeSelector := map[string]string{}
+
+					for k, v := range nodeselectorAddedByValues {
+						expectedNodeSelector[k] = v
+					}
+
+					for k, v := range defaultNodeSelector {
+						expectedNodeSelector[k] = v
+					}
+
+					assert.Equal(tc.T, expectedNodeSelector, podTemplateSpec.Spec.NodeSelector,
+						"workload %s (type: %T) does not have correct nodeSelectors, expected: %v got: %v",
+						obj.GetName(), obj, expectedNodeSelector, podTemplateSpec.Spec.NodeSelector,
+					)
+				}),
+			},
+		},
+
+		{
+			Name: "Check All Workload Have Tolerations As Per Given Values",
+			Covers: []string{
+				".Values.tolerations",
+			},
+
+			Checks: test.Checks{
+				checker.PerWorkload(func(tc *checker.TestContext, obj metav1.Object, podTemplateSpec corev1.PodTemplateSpec) {
+					if obj.GetName() == "kiali" {
```

Comment:
Use below method for selecting specific pods
```
if checker.Select("rancher-backup-patch-sa", "cattle-resources-system", sa) {
	return
}
```

Reply:
When using `checker.Select` the test is picking all the images including the images that sub-charts uses. And it's erroring out. That's why I used `obj.GetName()`

Reply:
I wrote it like this
```
if checker.Select("istioctl-installer", "istio-system", obj) {
	return
}